src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -376,11 +378,20 @@ def generate_python_code(
         pickled_func = pickle_function(func)
     args = ",".join(arg_names)
 
+    try:
+        source_code_comment = (
+            code_generation.generate_source_code(func) if source_code_display else ""
+        )
+    except Exception as exc:
+        logger.debug(f"Source code comment could not be generated due to error {exc!r}")
```

Comment:
I'm wondering if this could be part of telemetry that is sent over to snowhouse? Even some success/failure counter might be some useful signal.

Reply:
I have created a separate JIRA for this: https://snowflakecomputing.atlassian.net/browse/SNOW-651381 and would like to make a separate PR for it.

trying to restrict the scope of this PR.